### DISCUSSION TITLE
Enabling Secrets in Nextjs hosting deployment

### DISCUSSION THREAD

**Discussion by okaforcj** [firebase/firebase-tools#5860] [Q&A]
Hello, please is there a way to defineSecret for the Nextjs enabled deployments. For example when i deploy secretes for regular firebase functions, i can use defineSecret to prompt a secret that I want to add. Is this possible with the nextjs deployment pipeline or is there an easy work around?

**Comment by jamesdaniels** [comment]
We now allow you to [customize the backend](https://github.com/FirebaseExtended/firebase-framework-tools#configuring-your-backend) that is codegenerated for NextJS, etc. You can use the "secrets" configuration option.

**Reply by okaforcj** [reply]
thanks for your answer. i have implemented it using that configuration but i was wondering if it could prompt when the secret is not defined. with firebase functions deployment directly using defineSecret, it will prompt on the CLI to enter the secret value if it is not defined. with the frameworkBackendPipeline it will fail deployment completely if the value is not defined. I was just wondering if that feature specifically is available.
